src/rdkafka_topic.c

```diff
@@ -623,17 +623,22 @@ static int rd_kafka_toppar_leader_update (rd_kafka_topic_t *rkt,
 
         rd_kafka_toppar_lock(rktp);
 
-        if (leader != NULL &&
-            rktp->rktp_broker != NULL &&
-            rktp->rktp_broker->rkb_source != RD_KAFKA_INTERNAL &&
-            rktp->rktp_broker != leader &&
-            rktp->rktp_leader_id == leader_id) {
-                rd_kafka_dbg(rktp->rktp_rkt->rkt_rk, TOPIC, "BROKER",
-                        "Topic %s [%"PRId32"]: leader %"PRId32" unchanged, "
-                        "not migrating away from preferred replica %"PRId32,
-                        rktp->rktp_rkt->rkt_topic->str,
-                        rktp->rktp_partition,
-                        leader_id, rktp->rktp_broker_id);
+        if (rktp->rktp_leader_id == leader_id) {
+                rd_bool_t fetching_from_follower =
+                        leader != NULL &&
+                        rktp->rktp_broker != NULL &&
+                        rktp->rktp_broker->rkb_source != RD_KAFKA_INTERNAL &&
+                        rktp->rktp_broker != leader;
+
+                if (fetching_from_follower)
+                        rd_kafka_dbg(rktp->rktp_rkt->rkt_rk, TOPIC, "BROKER",
```

Comment:
first, this was (should be I hope?!) exactly the existing logic (reorganized). i've been reluctant to change any of this in case the slightest misunderstanding breaks fff...

anyway the wording of the debug message is only relevant to fff - and I feel good to have (specifically) because what's going on is more complicated, so more verbose logging is more important/useful. we could add additional logging if the leader doesn't change generally, but we've not done it before afaik.